Gordon360/Controllers/RecIMControllers/LeagueController.cs

```diff
@@ -0,0 +1,114 @@
+﻿using Gordon360.Models.CCT;
+using Gordon360.Services;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+
+namespace Gordon360.Controllers.RecIMControllers
+{
+    [Route("api/recim")]
+    public class LeagueController : GordonControllerBase
+    {
+        private readonly ILeagueService _leagueService;
+
+        public LeagueController(ILeagueService leagueService)
+        {
+            _leagueService = leagueService;
+        }
+
+        ///<summary>Gets a list of all Leagues</summary>
+        /// <returns>
+        /// All Existing Leagues 
+        /// </returns>
+        [HttpGet]
+        [Route("league")]
+        public ActionResult<IEnumerable<League>> GetAllLeagues()
+        {
+            var result = _leagueService.GetAllLeagues();
+
+            if (result == null)
+            {
+                return NotFound();
+            }
+            return Ok(result);
```

Comment:
This is a great example of when not to return a `NotFound` response. Firstly, `result` here will never be null because you're returning `_context.League.AsEnumerable()` from the service. So if there are no leagues, you'd get an empty enumerable, but never `null`.

More importantly, if there were no leagues, that doesn't mean that the service ran into an error trying to get a resource that doesn't exist (which is what 404 Not Found means), but rather that the server successfully got the resource and it was empty.

So, you can simplify this to:
```suggestion
            var leagues = _leagueService.GetAllLeagues();

            return Ok(leagues);
```

Reply:
got it, just changed them